tests/operator_test.go

```diff
@@ -457,20 +507,48 @@ spec:
 		deleteAllKvAndWait(ignoreDeleteOriginalKV)
 
 		kvs := getKvList()
-
 		if len(kvs) == 0 {
 			createKv(copyOriginalKv())
 		}
-		patchOperatorVersion(tests.KubeVirtVersionTag)
+
+		modified := patchOperatorVersion(originalOperatorVersion)
+		if modified {
+			// make sure we wait until redeploymemt started
+			waitForUpdateCondition(originalKv)
+		}
+
 		waitForKv(originalKv)
-		allPodsAreReady(tests.KubeVirtVersionTag)
+		allPodsAreReady(originalKv)
 
 		if workDir != "" {
 			err = os.RemoveAll(workDir)
 			workDir = ""
 			Expect(err).ToNot(HaveOccurred())
 		}
 
+		// make sure virt deployments use shasums again after each test
+		ensureShasums()
+
+	})
+
+	Describe("should start a VM", func() {
+		It("using virt-launcher with a shasum", func() {
+
+			By("starting a VM")
+			vmi := tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).To(BeNil())
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			By("getting virt-launcher")
+			uid := vmi.GetObjectMeta().GetUID()
+			labelSelector := fmt.Sprintf(v1.CreatedByLabel + "=" + string(uid))
+			pods, err := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).List(metav1.ListOptions{LabelSelector: labelSelector})
+			Expect(err).ToNot(HaveOccurred(), "Should list pods")
+			Expect(len(pods.Items)).To(Equal(1))
+			Expect(usesSha(pods.Items[0].Spec.Containers[1].Image)).To(BeTrue(), "launcher pod should use shasum")
```

Comment:
is it worth cross referencing that the expected sha is used too?

Reply:
Mh, I don't have direct access to the expected shasum here. And since we always only know 1 shasum per image, I think this is good enough?

Reply:
works for me